src/gui/src/search.cpp

```diff
@@ -257,10 +257,23 @@ void Search::updateShapes(odb::dbBlock* block)
   data.snet_via_shapes_.clear();
   data.snet_shapes_.clear();
 
+  LayerMap<std::vector<RouteBoxValue<odb::dbNet*>>> net_shapes;
+  LayerMap<std::vector<SNetValue<odb::dbNet*>>> snet_shapes;
+  LayerMap<std::vector<SNetSBoxValue<odb::dbNet*>>> snet_net_via_shapes;
   for (odb::dbNet* net : block->getNets()) {
-    addNet(net);
-    addSNet(net);
+    addNet(net, net_shapes);
+    addSNet(net, snet_shapes, snet_net_via_shapes);
   }
```

Comment:
The memory peak can be lowered by moving splitting this loop to only handle snet here and separately handle nets after ` snet_net_via_shapes.clear();`.  The extra loop over all nets should be minor in runtime.